### ISSUE TITLE
Running transpile synthesis translation method and basis ['r'] doesn't work

### ISSUE THREAD

**Issue by mtreinish** [Qiskit/qiskit-terra#5046]
### Information

- **Qiskit Terra version**: 0.20.0
- **Python version**: any
- **Operating system**: any

### What is the current behavior?

When running a transpilation with the `translation_method='synthesis'` raises an error

```
qiskit.transpiler.exceptions.TranspilerError: "It was not possible to use the basis ['r']"
```

### Steps to reproduce the problem

```python3
from qiskit import QuantumCircuit, transpile
qc = QuantumCircuit(1)
qc.u3(1, 2, 3, 0)
transpile(qc, basis_gates=['r'], translation_method='synthesis').draw()
```

### What is the expected behavior?

This works

### Suggested solutions

Fix it

**Comment by mtreinish** [comment]
So I think the hard failure will probably be fixed in #5216. Looking at the full traceback the error is coming from optimize 1q:

```
qiskit/transpiler/passes/optimization/optimize_1q_gates.py", line 210, in run
    raise TranspilerError('It was not possible to use the basis %s' % self.basis)
qiskit.transpiler.exceptions.TranspilerError: "It was not possible to use the basis ['r']"
```
This is coming from the fact the input circuit I'm using here has a u3 in it. It would probably not fail (but also not work correctly) if the input circuit was more than 1qubit or had gates that weren't u3, u2, or u1 in it.

That being said I think even after #5216 is merged this will essentially be the same bug as #5203. What's happening here is that synthesis translation method is failing to convert the u3 to r (which is what #5203 is about). Right now it's passing a circuit with a u3 to optimize1q which tries to reduce a u3 to a u1 or u2. But it fails because it doesn't understand r.  #5216 will address this case because it will recognize the r basis but it will still yield the incorrect circuit because it skips runs of length 1, so after it merges this will just return `u3(1, 2, 3)` as the circuit (which is the same bug as #5203).

Normally we don't hit this case because the basis translator will change u3 to r and then the optimize 1q pass becomes a no-op because it doesn't try to do anything with an r gate.

**Comment by kdk** [comment]
Closing as the error is resolved by #5216 , and final basis issue is a duplicate of #5203 .
